app/src/main/java/appricottsoftware/clarity/fragments/ChannelFragment.java

```diff
@@ -64,26 +101,349 @@ public void onAttach(Context context) {
+                //Log.i(TAG, "JSON CHANNEL RESPONSE" + response.toString());
+            }
+            @Override
+            public void onSuccess(int statusCode, Header[] headers, JSONArray response) {
+                super.onSuccess(statusCode, headers, response);
+                Log.e(TAG , "onSuccess: Array");
+                //Log.i(TAG, "JSON CHANNEL RESPONSE" + response.toString());
+
+                try {
+                    for (int i = 0; i < response.length(); i++) {
+                        JSONObject aChannel = response.getJSONObject(i);
+                        String title = aChannel.getString("title").toString();
+                        String imageURL = aChannel.getString("image").toString();
+                        RecyclerListItem newItem = new RecyclerListItem(title, imageURL);
+                        rListItems.add(newItem);
+                    }
+                } catch (JSONException e) {
+                    e.printStackTrace();
+                }
+                rAdapter = new RecyclerAdapter(rListItems, getContext(), true);
+                channelRecycler.setAdapter(rAdapter);
+            }
+            @Override
+            public void onSuccess(int statusCode, Header[] headers, String responseString) {
+                super.onSuccess(statusCode, headers, responseString);
+                Log.e(TAG, "");
+            }
+            @Override
+            public void onFailure(int statusCode, Header[] headers, Throwable throwable, JSONObject errorResponse) {
+                try {
+                    switch(statusCode) {
+                        case(0):
+                            Toast.makeText(getContext(),
+                                    "Server is down. Please try later.",
+                                    Toast.LENGTH_LONG).show();
+                            break;
+                        default:
+                            Log.i(TAG, "Channel onFailure. Default Switch. Status Code: " + statusCode);
+                            break;
+                    }
+                }
+                catch (Exception e) {
+                    e.printStackTrace();
+                }
+                super.onFailure(statusCode, headers, throwable, errorResponse);
+            }
+            @Override
+            public void onFailure(int statusCode, Header[] headers, Throwable throwable, JSONArray errorResponse) {
+                super.onFailure(statusCode, headers, throwable, errorResponse);
+                throwable.printStackTrace();
+                Log.e(TAG, "");
+            }
+
+            @Override
+            public void onFailure(int statusCode, Header[] headers, String responseString, Throwable throwable) {
+                super.onFailure(statusCode, headers, responseString, throwable);
+                throwable.printStackTrace();
+                Log.e(TAG, "");
+            }
+        });
+
+    }
+
+    private void searchAPI(String query) {
```

Comment:
This function might change from searching for episodes in the future.

Reply:
Is this the field the genres go into? I added a TODO so it wont be forgotten.